core/src/avm2/globals/vector.rs

```diff
@@ -249,12 +249,16 @@ pub fn concat<'gc>(
         return Err("Not a vector-structured object".into());
     };
 
+    let original_length = new_vector_storage.length();
+
+    let use_swf10_behavior = activation
+        .caller_movie()
```

Comment:
Pretty sure that this also is supposed to rely on declarer SWF (except for builtins, which propagate caller's context).
Can you document this with a `//TODO` here? I know we have more places like this already.

Reply:
`caller_movie` tracks the SWF that called the native function, and if it was called by another native function, it will use the new behavior, matching avmplus.